File: CarShopWeb/wwwroot/js/product.js
var dataTable;

$(document).ready(function () {
    loadDataTable();
});


function loadDataTable() {
    dataTable = $('#tblData').DataTable({
        "ajax": {
            "url": "/Admin/Product/GetAll"
        },
        "columns": [
            { "data": "make", "width": "5%" },
            { "data": "model", "width": "5%" },
            { "data": "description", "width": "10%" },
            { "data": "bodyType", "width": "5%" },
            { "data": "firstRegistration", "width": "3%" },
            { "data": "gear", "width": "5%" },
            { "data": "listPrice", "width": "5%" },
            { "data": "price", "width": "5%" },
            { "data": "vehicleType.name", "width": "5%" },
            { "data": "fuelType.name", "width": "5%" },
            {
                "data": "id",
                "render": function (data) {
                    return `
                            <div class="text-center">
                                <a href="/Admin/Product/Upsert/${data}" class="btn btn-success text-white" style="cursor:pointer">
                                    <i class="fas fa-edit"></i> 
                                </a>
                                <a onclick=Delete("/Admin/Product/Delete/${data}") class="btn btn-danger text-white" style="cursor:pointer">
                                    <i class="fas fa-trash-alt"></i> 
                                </a>
                            </div>
                           `;
                }, "width": "40%"
            }
        ]
    });
}

function Delete(url) {
    swal({
        title: "Are you sure you want to Delete?",
        text: "You will not be able to restore the data!",
        icon: "warning",
        buttons: true,
        dangerMode: true
    }).then((willDelete) => {
        if (willDelete) {
            $.ajax({
                type: "DELETE",
                url: url,
                success: function (data) {
                    if (data.success) {
                        toastr.success(data.message);
                        dataTable.ajax.reload();
                    }
                    else {
                        toastr.error(data.message);
                    }
                }
            });
        }
    });
}

/*{ "data": "make", "width": "10%" },*/
           /* { "data": "model", "width": "10%" },
 { "data": "description", "width": "10%" },*/
/*   { "data": "bodyType", "width": "5%" },
   { "data": "firstRegistration", "width": "5%" },
   { "data": "listPrice", "width": "5%" },
   { "data": "gear", "width": "5%" },
   { "data": "vehicleType.name", "width": "10%" },
   { "data": "fuelType.name", "width": "5%" },*/

File: CarShop.Models/Product.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text;

namespace CarShop.Models
{
    public class Product
    {
        [Key]
        public int Id { get; set; }
        [Required]
        public string Make { get; set; }
        [Required]
        public string Model { get; set; }
        public string Description { get; set; }
        
        [Required]
        public string BodyType { get; set; }
        [Required]
        public int FirstRegistration { get; set; }
        [Required]
        [Range(1, 10000)]
        public double ListPrice { get; set; }
        [Required]
        public string Gear { get; set; }
        [Required]
        [Range(1, 10000)]
        public double Price { get; set; }
        
        public string ImageUrl { get; set; }

        [Required]
        public int VehicleTypeId { get; set; }
        
        [ForeignKey("VehicleTypeId")]
        public VehicleType VehicleType { get; set; }

        [Required]
        public int FuelTypeId { get; set; }

        [ForeignKey("FuelTypeId")]
        public FuelType FuelType { get; set; }
    }
}
